x-pack/plugin/security/lib/njj-remapped-intermediate/build.gradle

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+apply plugin: 'elasticsearch.build'
+apply plugin: 'com.github.johnrengelman.shadow'
+
+dependencies {
+  implementation "com.nimbusds:nimbus-jose-jwt:9.37.3"
+}
+
+tasks.named('shadowJar').configure {
+  // Attempting to exclude all of the classes we *don't* move here ought to be possible per the
+  // shadowJar docs, but actually attempting to do so results in an empty JAR. So we'll do that as a
+  // separate step (see njj-moved-utils-only)
```

Comment:
Seriously, I spent some quality time with the shadowJar source trying to figure out why but all the include-based strategies to define which files to include resulted in an empty JAR, and exclude-based ones would require listing out a bunch of files individually since there's a bunch of other stuff in the `com.nimbusds.jose.util.JSONStringUtils` package. I still don't understand why; I think there must be some bug with the relocate implementation and include directives. Breaking it out into two steps like this solves the problem relatively cleanly, but I welcome anyone who can figure out the correct incantation to do the right thing in one step - I think it's possible.

Reply:
To clarify, `exclude 'com/nimbus/**'` also results in an empty JAR, even though it's exclude-based, regardless of where it's put in the process.